packages/lesswrong/server/fmCrosspost/denormalizedFields.ts

```diff
@@ -0,0 +1,31 @@
+import { hasStringParam, hasBooleanParam } from "./validationHelpers";
+import { pick } from "underscore";
+
+/**
+ * In general, we try to keep a single source of truth for all post data that's
+ * crossposted on the original server and let the foreign server make graphql
+ * requests when it needs access to this.
+ *
+ * Some fields have to be denormalized across sites and these are defined here. In
+ * general, a field needs to be denormalized if it's used by PostsList2 or
+ * in database selectors (but these rules aren't strict).
+ *
+ * When adding a new field here, make sure to also update isValidDenormalizedData
+ */
+export const denormalizedFieldKeys = [
+  "draft",
+  "deletedDraft",
+  "title",
+  "isEvent",
+] as const;
+
+export const isValidDenormalizedData = (payload: unknown): payload is DenormalizedCrosspostData =>
+  hasBooleanParam(payload, "draft") &&
+  hasBooleanParam(payload, "deletedDraft") &&
+  hasStringParam(payload, "title") &&
+  hasBooleanParam(payload, "isEvent");
```

Comment:
I threw an && false here and got `Failed to update crosspost draft status` with no other context. It wouldn't let me update the post at all. Possibly not a new error though.